### DISCUSSION TITLE
Getting a 404 GET Error when enabling comments

### DISCUSSION THREAD

**Discussion by zerenxyz** [squidfunk/mkdocs-material#4630] [Q&A] [answered]
I followed the docs for [enabling comments with Giscus](https://squidfunk.github.io/mkdocs-material/setup/adding-a-comment-system/#giscus-integration) and I added the snippet of code to the comments.html partials folder in overrides like so:
```js
{% if page.meta.comments %}
  <h2 id="__comments">{{ lang.t("meta.comments") }}</h2>
  <script src="https://giscus.app/client.js"
        data-repo="zerenxyz/comments"
        data-repo-id="..."
        data-category-id="..."
        data-mapping="pathname"
        data-strict="0"
        data-reactions-enabled="1"
        data-emit-metadata="0"
        data-input-position="bottom"
        data-theme="light"
        data-lang="en"
        crossorigin="anonymous"
        async>
</script>

  <!-- Synchronize Giscus theme with palette -->
  <script>
    var giscus = document.querySelector("script[src*=giscus]")

    /* Set palette on initial load */
    var palette = __md_get("__palette")
    if (palette && typeof palette.color === "object") {
      var theme = palette.color.scheme === "slate" ? "dark" : "light"
      giscus.setAttribute("data-theme", theme) 
    }

    /* Register event handlers after documented loaded */
    document.addEventListener("DOMContentLoaded", function() {
      var ref = document.querySelector("[data-md-component=palette]")
      ref.addEventListener("change", function() {
        var palette = __md_get("__palette")
        if (palette && typeof palette.color === "object") {
          var theme = palette.color.scheme === "slate" ? "dark" : "light"

          /* Instruct Giscus to change theme */
          var frame = document.querySelector(".giscus-frame")
          frame.contentWindow.postMessage(
            { giscus: { setConfig: { theme } } },
            "https://giscus.app"
          )
        }
      })
    })
  </script>
{% endif %}
```

And I enabled it on a page to test, but I get the following error when I try and open the page:
![_H1CYwEx496](https://user-images.githubusercontent.com/52886201/201499041-06a2d85c-db6e-442b-a1f7-04b97af36f3b.png)
Not sure if the `GET` url has any unique ID's so I blurred it, but I can post it if needed. When going to the page with the comments it loads an iFrame of the project itself with `404 - Not found`

**Comment by squidfunk** [comment, marked as answer]
In general, it's impossible to help you without exactly knowing your setup... That being said, the request seems to be local – are you using [the built-in privacy plugin](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#built-in-privacy-plugin)? Giscus can't be automatically self-hosted by the privacy plugin, because it's an SPA. [Giscus and MathJax must be excluded](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#external-assets) via [`external_assets_exclude`](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#+privacy.external_assets_exclude):

``` yaml
plugins:
  - privacy:
      external_assets_exclude: 
        - cdn.jsdelivr.net/npm/mathjax@3/* 
        - giscus.app/*
```

If that's not the case, please, as always, provide a _minimal reproducible example_... Otherwise, we can only guess and that eats time.

**Reply by zerenxyz** [reply]
My bad for not providing a minimal reporudcible example.

I am using the built-in privacy plugin and adding the exclude suggestion you linked fixed the issue for me.
